Remotion/ObjectBinding/Web.Test/IndividualControlTests/BocListUserControl.ascx.cs

```diff
@@ -392,6 +400,186 @@ private void ChildrenListRemoveRowsButton_Click (object sender, EventArgs e)
+                    },
+                    "Bad last job dates",
+                    "Localized bad last job dates"),
+                // Children property failures
+                new PropertyValidationFailure(
+                    person,
+                    childrenProperty,
+                    person.Children,
+                    "Bad children",
+                    "Localized bad children"),
+                // First child failure (visible)
+                new ObjectValidationFailure(
+                    firstChild,
+                    "Bad first child",
+                    "Localized bad first child"),
+                new PropertyValidationFailure(
+                    firstChild,
+                    personPartnerProperty,
+                    firstChild.Partner,
+                    "Bad first child.Partner",
+                    "Localized bad first child.Partner"),
+                new ObjectValidationFailure(
+                    firstChild,
+                    new[]
+                    {
+                        new ValidatedProperty(firstChild, personLastNameProperty),
+                    },
+                    "Bad first child name",
+                    "Localized bad first child names"),
+                new ObjectValidationFailure(
+                    firstChild,
+                    new[]
+                    {
+                        new ValidatedProperty(firstChild, personLastNameProperty),
+                        new ValidatedProperty(firstChild, personFullNameProperty)
+                    },
+                    "Bad first child names",
+                    "Localized bad first child names"),
+                // Last child failure (invisible)
+                new ObjectValidationFailure(
+                    lastChild,
+                    "Bad last child",
+                    "Localized bad last child"),
+                new PropertyValidationFailure(
+                    lastChild,
+                    personPartnerProperty,
+                    lastChild.Partner,
+                    "Bad last child.Partner",
+                    "Localized bad last child.Partner"),
+                new ObjectValidationFailure(
+                    lastChild,
+                    new[]
+                    {
+                        new ValidatedProperty(lastChild, personLastNameProperty),
+                    },
+                    "Bad last child name",
+                    "Localized bad last child name"),
+                new ObjectValidationFailure(
+                    lastChild,
+                    new[]
+                    {
+                        new ValidatedProperty(lastChild, personLastNameProperty),
+                        new ValidatedProperty(lastChild, personFullNameProperty)
+                    },
+                    "Bad last child names",
+                    "Localized bad last child names"),
+            }));
+
+    PrepareValidation();
+    FormGridManager.PrepareValidation();
```

Comment:
I don't think we should have to use FormGridManager.PrepareValidation(). I don't use it in other places, either, so it should be covered by this.PrepareValidation(). What we are missign is Page.PrepareValidation() since you're doing this from inside the UserControl. I would recommend trying

```suggestion
    ((ISmartPage)Page).PrepareValidation();
```

Reply:
I could not make this work. Found no method called prepareValidation on ISmartPage or anything similar to that.

Reply:
Hmm...please check Form.aspx.cs, line 105. That's the one we want to call.